src/nfts/NftGallery.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { RefreshControl, ScrollView, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import Touchable from 'src/components/Touchable'
+import ImageErrorIcon from 'src/icons/ImageErrorIcon'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import NftImage from 'src/nfts/NftImage'
+import NftsLoadError from 'src/nfts/NftsLoadError'
+import { myNftsErrorSelector, myNftsLoadingSelector, myNftsSelector } from 'src/nfts/selectors'
+import { fetchMyNfts } from 'src/nfts/slice'
+import { Nft, NftOrigin } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+
+// This is used instead of a gap between the images because the gap is not supported in RN yet.
+const imageSize = Math.min(variables.width / 2.5, variables.height / 4)
+
+function NftTouchable({ nft }: { nft: Nft }) {
```

Comment:
nit: this is my personal opinion but i don't really like having multiple components declared in the same file. it makes the file cluttered and difficult to read. so i think this abstraction is unneeded, it seems like we could have this code inside the main gallery component without complicating that component too much and reduce the amount of abstraction